### ISSUE TITLE
Support type inference for default value parameters in lambda functions

### ISSUE THREAD

**Issue by kennipj** [microsoft/pyright#6558]
**Is your feature request related to a problem? Please describe.**
Lambda functions that have parameters with default values do not currently have their type inferred. 
Example:
```py
lambda_func = lambda x="test": reveal_type(x) # Type of "x" is "Unknown"
reveal_type(lambda_func) # Type of "lambda_func" is "(x: Unknown = "test") -> Unknown"
```

This is in contrast to named functions:
```py
def named_func(x="test"):
    return reveal_type(x)  # Type of "x" is "str"

reveal_type(named_func)  # Type of "named_func" is "(x: str = "test") -> str"
```

This discrepancy reduces type safety inside the body of the lambda function, as well as for the function callers, as the return type may be unknown in some cases.

**Describe the solution you’d like**
Ideally lambda functions with default parameter values perform the same type inference as named functions.

Tested on pyright 1.1.337

**Comment by erictraut** [comment]
Thanks for the enhancement request.

Default argument values are unusual for lambdas, as are situations where a lambda appears without additional type context to enable bidirectional type inference. I suspect that's why no one has suggested this feature previously.

I think it's reasonable to apply the same inference logic that pyright uses for unannotated parameters with default arguments within named functions.

This functionality will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.338, which I just published. It will also be included in a future release of pylance.
